internal/controller/common/rollout_controller.go

```diff
@@ -170,6 +170,9 @@ func UpdateUpgradeState(ctx context.Context, c client.Client, upgradeState commo
 		childObject.GetKind(), childObject.GetNamespace(), childObject.GetName())
 	var patchJson string
 	labels := childObject.GetLabels()
+	if labels == nil {
```

Comment:
This is probably unnecessary but was unsure if this would help with the unit test failures.

Reply:
this seems good just in case